### PR TITLE
Issue 6 move work pool zk subtree to distinct place

### PR DESCRIPTION
**Pull request by KazankovMarch**
Blocked by https://github.com/ru-fix/distributed-job-manager/pull/25
See https://github.com/ru-fix/distributed-job-manager/issues/6

Editing `work-pool` tree goes only  after checking version of `work-pool-version` node. `registration-version` renamed to `worker-version`. Editing `workers/worker` and `alive` subtrees goes only after checking version of `worker-version` node.
So, when worker reconnected, it should check versions of the both these nodes to register itself as alive and register its jobs.
`assignment-version` renamed to `worker-assignment-version` to express that `worker-version` is "above" `worker-assignment-version` (as in zk tree).
See updated docs.

### CODE REVIEW

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java [hunk 1]
```diff
@@ -244,48 +238,65 @@ private void registerWorkerAsAliveAndRegisterJobs(
         workPooledCache.start();
     }
 
+    /**
+     * Add in zk paths of new work items, that contains in newWorkPool, but doesn't in currentWorkPool and
+     * remove work items, that contains in currentWorkPool, but doesn't in newWorkPool.
```
Comment (swarmshine) [inline distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java]: convert comments to two functions and call them from mergeWorkPoolsForJob

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java [hunk 2]
```diff
@@ -244,48 +238,65 @@ private void registerWorkerAsAliveAndRegisterJobs(
...
+     */
+    private boolean mergeWorkPoolsForJob(
```
Comment (swarmshine) [inline distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java]: seems like this is not merge but updateZkJobWorkPool

File: distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java (hunk 1 above)
Comment (KazankovMarch) [reply distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java]: :thinking:

File: asciidoc/index.adoc [hunk 3]
```diff
@@ -124,52 +124,102 @@ Main goal of rebalance process is to minimize unnecessary job restarts and reass
...
+Worker listens for updates in `workers/worker/assigned` subtree and stops/launches jobs accordingly. +
+Worker also periodically updates its jobs in `work-pool` subtree using `work-pool-version` lock +
+
+Manager life-cycle:
 
+Manager listens for updates in `alive` subtree and invokes rebalance if set of alive workers was changed: +
 
+* Updates workers `assigned` subtree using `worker-assignment-version` lock
+
+In future releases:
```
Comment (KazankovMarch) [inline asciidoc/index.adoc]: probably in next pull-request

**Pull request merged**